File: Difficulty: Medium/Form the Largest Number/form-the-largest-number.cpp
class Solution {
  public:
    static bool comparator(string x, string y) {
        string xy = x + y;
        string yx = y + x;
        return xy > yx;
    }
    
    string findLargest(vector<int> &arr) {
        vector<string> nums;
        for(int num : arr) nums.push_back(to_string(num));
        
        sort(nums.begin(), nums.end(), comparator);
        
        string ans = "";
        for(int i=0; i<nums.size(); i++)
            ans += nums[i];
            
        return ans[0] == '0' ? "0" : ans;
    }
};

File: Difficulty: Medium/Form the Largest Number/form-the-largest-number.java
class Solution {
    public String findLargest(int[] arr) {
        String[] nums=new String[arr.length];
        StringBuilder largestNumber=new StringBuilder();
        int i=0;
        for(int x:arr){
            nums[i++]=String.valueOf(x);
        }
        Arrays.sort(nums,(a,b)->(b+a).compareTo(a+b));
        if(nums[0].charAt(0)=='0'){
            return "0";
        }
        for(String x:nums){
            largestNumber.append(x);
        }
        return largestNumber.toString();
    }
}